compiler-cli/src/http.rs

```diff
@@ -42,3 +43,32 @@ impl gleam_core::io::HttpClient for HttpClient {
             .map_err(Error::http)
     }
 }
+
+fn init_client() -> reqwest::Client {
+    if let Some(cert) = get_certificate() {
+        return reqwest::Client::builder()
+            .add_root_certificate(cert)
+            .build()
+            .expect("Unable to initialize a reqwest HTTP client");
+    } else {
+        return reqwest::Client::new();
+    }
+}
+
+fn get_certificate() -> Option<reqwest::Certificate> {
+    match std::env::var(CERTS_ENV_VAR) {
+        Ok(certs_path) => {
+            let data = std::fs::read(certs_path).expect(&format!(
+                "Unable to read certs file set as `{}`",
+                CERTS_ENV_VAR
+            ));
+            let cert = reqwest::Certificate::from_pem(&data).expect(&format!(
+                "Unable to construct a certificate from certs file set as `{}`",
+                CERTS_ENV_VAR
+            ));
```

Comment:
The two `expect`s here cause the build tool to crash in the event that the user provides an invalid path or certs. We don't crash on invalid input, so instead errors will need to be returned from this function.

Our file io module has a file read function which will do this for you, and you'll need to construct our file IO error manually for the parsing of the certs.